core/chains/evm/gas/models.go

```diff
@@ -75,24 +85,24 @@ func NewEstimator(lggr logger.Logger, ethClient evmclient.Client, cfg Config, ge
 	switch s {
 	case "Arbitrum":
 		newEstimator = func(l logger.Logger) EvmEstimator {
-			return NewArbitrumEstimator(lggr, geCfg, ethClient, ethClient)
+			return NewArbitrumEstimator(lggr, geCfg, ethClient)
 		}
 	case "BlockHistory":
 		newEstimator = func(l logger.Logger) EvmEstimator {
 			return NewBlockHistoryEstimator(lggr, ethClient, cfg, geCfg, bh, *ethClient.ConfiguredChainID())
 		}
 	case "FixedPrice":
 		newEstimator = func(l logger.Logger) EvmEstimator {
-			return NewFixedPriceEstimator(geCfg, bh, lggr)
+			return NewFixedPriceEstimator(geCfg, ethClient, bh, lggr, cfg.ChainType())
 		}
 	case "L2Suggested", "SuggestedPrice":
 		newEstimator = func(l logger.Logger) EvmEstimator {
-			return NewSuggestedPriceEstimator(lggr, ethClient, geCfg)
+			return NewSuggestedPriceEstimator(lggr, ethClient, geCfg, cfg.ChainType())
 		}
 	default:
 		lggr.Warnf("GasEstimator: unrecognised mode '%s', falling back to FixedPriceEstimator", s)
 		newEstimator = func(l logger.Logger) EvmEstimator {
-			return NewFixedPriceEstimator(geCfg, bh, lggr)
+			return NewFixedPriceEstimator(geCfg, ethClient, bh, lggr, cfg.ChainType())
 		}
 	}
 	return NewWrappedEvmEstimator(lggr, newEstimator, df, l1Oracle, geCfg)
```

Comment:
Might be out of scope, but **why** do we pass a function that creates an estimator instead of the actual estimator 🤔 ?

Reply:
Side note: we have `EvmFeeEstimator`,  `EvmEstimator`, `WrappedEvmEstimator` which is quite confusing. `WrappedEvmEstimator` seems like the implementation of `EvmFeeEstimator` interface and its scope is local. Probably we can rename it to `evmFeeEstimator` and make it private, so by looking you can quickly understand the association.

Reply:
> but why do we pass a function that creates an estimator instead of the actual estimator 🤔
I'm not sure, but this is preexisting functionality so I believe it's out of scope for this PR

Reply:
> Probably we can rename it to evmFeeEstimator and make it private, so by looking you can quickly understand the association.

Agreed, will implement this change